redis/src/main/scala/zio/redis/options/Server.scala

```diff
@@ -0,0 +1,118 @@
+      case s"!$password"            => Some(RemoveHashedPassword(password))
+      case "reset"                  => Some(Reset)
+      case _                        => None
+    }
+
+    case object On                                                                   extends Rule
+    case object Off                                                                  extends Rule
+    sealed case class KeyPattern(pattern: String)                                    extends Rule
+    case object ResetKeys                                                            extends Rule
+    sealed case class ChannelPattern(pattern: String)                                extends Rule
+    case object RestChannels                                                         extends Rule
+    sealed case class AddCommand(command: String, subcommand: Option[String] = None) extends Rule
+    sealed case class RemoveCommand(command: String)                                 extends Rule
+    sealed case class AddCategory(category: String)                                  extends Rule
+    sealed case class RemoveCategory(category: String)                               extends Rule
+    case object NoPass                                                               extends Rule
+    sealed case class AddTextPassword(pw: String)                                    extends Rule
+    sealed case class RemoveTextPassword(pw: String)                                 extends Rule
+    sealed case class AddHashedPassword(hash: String)                                extends Rule
+    sealed case class RemoveHashedPassword(hash: String)                             extends Rule
+    case object Reset                                                                extends Rule
+
+    final val AllKeys: KeyPattern         = KeyPattern("*")
+    final val AllChannels: ChannelPattern = ChannelPattern("*")
+    final val AllCommands: AddCommand     = AddCommand("all")
+    final val NoCommands: RemoveCommand   = RemoveCommand("all")
+  }
+
+  sealed case class UserInfo(
+    flags: Chunk[String],
+    passwords: Chunk[String],
+    commands: String,
+    keys: Chunk[String],
+    channels: Chunk[String]
+  )
+
+  sealed case class UserEntry(username: String, rules: Chunk[Rule])
+
+  object UserEntry {
+    def unapply(value: String): Option[UserEntry] = value match {
+      case s"user $username $rulesString" =>
+        Chunk
+          .fromArray(rulesString.split(' '))
+          .forEach(Rule.unapply)
+          .map(rules => UserEntry(username, rules))
+
+      case _ =>
+        None
+    }
+  }
+
+  sealed case class LogEntry(
+    count: Int,
+    reason: String,
+    context: String,
+    `object`: String,
+    username: String,
+    ageDuration: Duration,
+    clientInfo: String
+  )
+
+  sealed case class CommandDetail(
+    name: String,
+    artity: Int,
+    flags: Chunk[String],
+    firstKey: Int,
+    lastKey: Int,
+    step: Int,
```

Comment:
I think, We have to stick with Long type everywhere, cause Redis integer is 64 bit in most cases, according to documentation [Resp Integer](https://redis.io/topics/protocol#resp-integers)